### PR TITLE
Board & solver

### PR DESCRIPTION
**Pull request by zetraxus**

### CODE REVIEW

File: src/data_structure/Board.java [hunk 1]
```diff
@@ -84,32 +83,20 @@ private void fillGameBoard(Field2D[][] initialBoard) {
...
                 if (gameBoard[j][i].getType() == Field2D.Type.WRITABLE) {
                     if (shortcut.charAt(indexInShortcut) != ASCIZEROCODE)
-                        this.setField(j, i, (int) shortcut.charAt(indexInShortcut) - ASCIZEROCODE);
+                        gameBoard[j][i].getWritable().setValue((int) shortcut.charAt(indexInShortcut) - ASCIZEROCODE);
                     ++indexInShortcut;
                 }
             }
         }
-        //setPossibilities(); // TODO: is this needed or not?
+        setPossibilitiesBasedOnTemplate();
```
Comment (kam193) [inline src/data_structure/Board.java]: Is this needed? Łukasz said on previous PR, that we don't need setting possibilities here

File: src/data_structure/Board.java [hunk 2]
```diff
@@ -275,43 +235,49 @@ private void setPossibilities() {
...
         }
-        for (FieldWritable fieldInRow : gameBoard[x][y].getWritable().getRowFieldInfo().getFields()) {
-            if (fieldInRow != gameBoard[x][y].getWritable() && fieldInRow.getState() == FieldWritable.State.FILLED) {
-                gameBoard[x][y].getWritable().revokePossibility(fieldInRow.getValue());
+        for (FieldWritable fieldInRow : writable.getRowFieldInfo().getFields()) {
+            updatePossibilitiesInField(writable, fieldInRow);
+        }
+    }
+
+    private void updatePossibilitiesInField(FieldWritable writable, FieldWritable fieldInRow) {
```
Comment (kam193) [inline src/data_structure/Board.java:284]: Here `fieldInRow`, on L268 `fieldInColumn`, in fact, what is it? Mayby name should be more generic

File: src/sample/Main.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 public class Main extends Application {
 
     public static void main(String[] args) {
-        String inputFile = "examples/example.in";
+        String inputFile = "examples/board_8x8_hard.slow";
```
Comment (kam193) [inline src/sample/Main.java]: I think, we shouldn't commit changes in Main.java that are used only for testing/running app for debug.

File: src/solver/Solver.java [hunk 4]
```diff
@@ -54,9 +51,14 @@ public Board solve() {
 
             Vector<GameState> newGeneratedStates = analyzedBoard.nextStep();
             for (int i = 0; i < newGeneratedStates.size(); ++i) {
-                queue.add(newGeneratedStates.elementAt(i));
-                System.out.println("" + analyzed.isSolved() + " " + newGeneratedStates.elementAt(i).getBoardShortcut());
+                if(!mapsAddedToQueue.contains(newGeneratedStates.elementAt(i).getBoardShortcut())){
```
Comment (kam193) [inline src/solver/Solver.java]: Maybe will be better change iterating like `for (GameState state : newGeneratedStates)`, we don't use index in any other way that to get element from vector.

File: src/solver/Solver.java [hunk 5]
```diff
@@ -54,9 +51,14 @@ public Board solve() {
...
+                    queue.add(newGeneratedStates.elementAt(i));
+                    mapsAddedToQueue.add(newGeneratedStates.elementAt(i).getBoardShortcut());
+//                    System.out.println("" + analyzed.isSolved() + " " + newGeneratedStates.elementAt(i).getBoardShortcut());
```
Comment (kam193) [inline src/solver/Solver.java]: Please delete it if printing is no longer helpful :)

File: src/data_structure/Board.java (hunk 1 above)
Comment (zetraxus) [reply src/data_structure/Board.java]: i changed line 93 so we need it

**Pull request merged**